DRILL-8276: Add Support for User Translation for Splunk Plugin

cgivre opened the pull request:
# [DRILL-8276](https://issues.apache.org/jira/browse/DRILL-8276): Add Support for User Translation for Splunk Plugin

## Description
Adds support for user translation in the Splunk plugin.

## Documentation
Not really much to add here.  Splunk plugin now supports user translation in a similar manner as the JDBC plugin. 

## Testing
Added unit tests, tested manually.

cgivre commented:
@jnturton This PR is now ready for review.  As a note, I couldn't get this to work on my Mac M1 due to the lack of a ARM Docker Container for Splunk.  Not sure how the Travis CI checks are passing.

jnturton commented:
It looks like no attempt was made to run the Splunk tests on ARM. Perhaps this is because we separated out the running of the SlowTest category? @luocooong, do we run any SlowTests on ARM?

```
[INFO] --- maven-surefire-plugin:3.0.0-M7:test (default-test) @ drill-storage-splunk ---
[INFO] Tests are skipped.
```

cgivre commented:
> It looks like no attempt was made to run the Splunk tests on ARM. Perhaps this is because we separated out the running of the SlowTest category? @luocooong, do we run any SlowTests on ARM?
> 
> ```
> [INFO] --- maven-surefire-plugin:3.0.0-M7:test (default-test) @ drill-storage-splunk ---
> [INFO] Tests are skipped.
> ```

Just to be clear about it, there apparently is no working Docker container for Splunk on ARM.  I did a lot of searching for this and all I found was a bunch of angry developer threads on Splunk's help boards[1] [2]. (There are more but you get the idea) 

I was able to start Splunk enough to run queries on my M1, but I couldn't get a containerized version to work at all.

I'm not too concerned because we know Drill runs on ARM, and the use case here would be for a user to connect to a remote Splunk installation.

[1]: https://github.com/splunk/docker-splunk/issues/493
[2]: https://community.splunk.com/t5/Splunk-Enterprise/Splunk-Web-on-Docker-fails-to-start-using-an-M1-Mac-on-Monterey/m-p/581983

jnturton commented:
> I'm not too concerned because we know Drill runs on ARM, and the use case here would be for a user to connect to a remote Splunk installation.

I see it the same way, I just want to shed light on what's happening in Travis. I don't believe it's a concern for this PR.

jnturton commented on contrib/storage-splunk/src/main/java/org/apache/drill/exec/store/splunk/SplunkGroupScan.java:
```diff
@@ -50,17 +54,20 @@ public class SplunkGroupScan extends AbstractGroupScan {
 
   private int hashCode;
 
+  private MetadataProviderManager metadataProviderManager;
```
Can you elaborate on the role played by the metastore in this PR a bit, please?

cgivre replied:
This actually isn't related to the metastore.  It is some boilerplate code to allow for provided schema.  You have to have  the MetadataProviderManager to access the inline schemata, which you then add to the schema negotiator in the ScanBatchCreator class.  This also is why I added all of the `getPhysicalScan()` functions to the `SplunkStoragePlugin` class.   With that said, this PR does not go all the way and add provided schema support.  I will do that as a separate PR.

For future storage plugins, we should make that a standard, as it really isn't difficult to add.

jnturton commented on contrib/storage-splunk/src/test/java/org/apache/drill/exec/store/splunk/TestSplunkUserTranslation.java:
```diff
@@ -0,0 +1,80 @@
+    String sql = "SHOW DATABASES";
+
+    RowSet results = client.queryBuilder().sql(sql).rowSet();
+    assertEquals(8, results.rowCount());
```
I guess something like this would take care of it: `show databases where table_schema like 'ut_splunk.%'` and `assertEquals(0, results.rowCount());`

cgivre replied:
Fixed

jnturton commented on contrib/storage-splunk/src/test/java/org/apache/drill/exec/store/splunk/TestSplunkUserTranslation.java:
```diff
@@ -0,0 +1,80 @@
+    String sql = "SHOW DATABASES";
+
+    RowSet results = client.queryBuilder().sql(sql).rowSet();
+    assertEquals(9, results.rowCount());
```
Restrict this count to schemas registered by _this_ plugin using a LIKE filter in the SHOW DATABASES, or similar.

cgivre commented:
@jnturton Thanks for the review.  I revised the unit tests and added an additional test to verify that a query will fail with invalid credentials.  I also fixed the LGTM alert.

jnturton reviewed (approved):
Thanks, LGTM.
